packages/react-component-library/src/components/Select/Select.test.tsx

```diff
@@ -333,4 +333,33 @@ describe('Select', () => {
       })
     })
   })
+
+  describe('when the optional `onChange` callback is not provided and the user selects an items', () => {
+    beforeEach(() => {
+      wrapper = render(
+        <Select
+          options={[
+            ...options,
+            {
+              label: 'Foo',
+              value: 'foo',
+            },
+          ]}
+        />
+      )
+
+      const input = wrapper.getByTestId('react-select-vendor-input')
+      fireEvent.focus(input)
+      fireEvent.keyDown(input, {
+        key: 'ArrowDown',
+        code: 40,
+      })
+    })
+
+    it('should not attempt to invoke the `onChange` callback', () => {
+      expect(() => userEvent.click(wrapper.getByText('Foo'))).not.toThrow(
```

Comment:
I am wondering if checking the message is quite specific and just having `.not.toThrow()` is enough but actually it just works as it is so up to you.